RAGens/win32/MakeBuildVer.bat

```diff
@@ -1,15 +1,33 @@
 @echo off
 
-git describe --tags --match "RAGens.*" > LiveTag.txt
-@set /p ACTIVE_TAG=<LiveTag.txt
-@set VERSION_NUM=%ACTIVE_TAG:~7,3%
-
-git diff HEAD > Diffs.txt
-@set /p RAW_DIFFS_FOUND=<Diffs.txt
+git describe --tags --match "RAGens.*" > Temp.txt
+set /p ACTIVE_TAG=<Temp.txt
+set VERSION_NUM=%ACTIVE_TAG:~7,3%
+set VERSION_REVISION=%ACTIVE_TAG:~11,-9%
+if "%VERSION_REVISION%"=="" set VERSION_REVISION=0
 
 setlocal
-@for /F "usebackq" %%A in ('"Diffs.txt"') do set DIFF_FILE_SIZE=%%~zA
-@if %DIFF_FILE_SIZE% GTR 0 set ACTIVE_TAG=Unstaged Changes
+git diff HEAD > Temp.txt
+for /F "usebackq" %%A in ('"Temp.txt"') do set DIFF_FILE_SIZE=%%~zA
+if %DIFF_FILE_SIZE% GTR 0 (
+    set ACTIVE_TAG=Unstaged changes
+    set VERSION_MODIFIED=1
+) else (
+    set VERSION_MODIFIED=0
+)
 
 @echo Tag: %ACTIVE_TAG% (%VERSION_NUM%)
-@echo #define RAGENS_VERSION "0.%VERSION_NUM%" > ../common/src/BuildVer.h
\ No newline at end of file
+@echo #define RAGENS_VERSION "0.%VERSION_NUM%.%VERSION_REVISION%.%VERSION_MODIFIED%" > BuildVer2.h
```

Comment:
support for full version info:
`0.XXX.YYY.ZZZ`
where `XXX` is the tag version, `YYY` is the number of commits since the tag, and `ZZZ` is 1 if local modifications exist. This value will be passed to RA_Integration, which will pass it on to the server for possible custom build detection.